code/game/objects/structures/crates_lockers/closets/secure/genpop.dm

```diff
@@ -18,6 +18,11 @@
 
 	return ..()
 
+/obj/structure/closet/secure_closet/genpop/handle_lock_addition() //If lock construction is successful we don't care what access the electronics had, so we override it
+	if(..())

```

Comment:
The genpop lockers are still broken as shit. They're not meant to be special in any inherent way; they're just hunks of metal with special electronics attached. The electronics attached should just be a somehow modified version of airlock electronics, not the typepath of the locker. That will be nonsensical.